builtin/providers/aws/structure.go

```diff
@@ -813,11 +813,20 @@ func flattenStepAdjustments(adjustments []*autoscaling.StepAdjustment) []map[str
 	return result
 }
 
-func flattenResourceRecords(recs []*route53.ResourceRecord) []string {
+func flattenResourceRecords(recs []*route53.ResourceRecord, typeStr string) []string {
 	strs := make([]string, 0, len(recs))
 	for _, r := range recs {
 		if r.Value != nil {
-			s := strings.Replace(*r.Value, "\"", "", 2)
+			s := *r.Value
+			if typeStr == "TXT" || typeStr == "SPF" {
+				// On Route53, these records are usually quoted.  This plugin defines the
+				// canonical format to be unquoted.  So when converting from the Route53 value
+				// to the Terraform config value, remove quotes.
+				l := len(s)
+				if l > 1 && s[0] == '"' && s[l-1] == '"' {
+					s = s[1:(l - 1)]
+				}
```

Comment:
Apologies if this will sound like nitpicking... wouldn't [`strings.Trim`](https://golang.org/pkg/strings/#Trim) make this look less complex and cleaner?